homeassistant/config_entries.py

```diff
@@ -1528,6 +1531,99 @@ class OptionsFlow(data_entry_flow.FlowHandler):
     handler: str
 
 
+class HelperCommonFlowHandler:
+    """Handle a config or options flow for helper."""
+
+    def __init__(
+        self,
+        handler: HelperConfigFlowHandler | HelperOptionsFlowHandler,
+        config_entry: ConfigEntry | None,
+    ) -> None:
+        """Initialize a common handler."""
+        self._handler = handler
+        self._options = dict(config_entry.data) if config_entry is not None else {}
+
+    @staticmethod
+    async def validate_input(
+        hass: HomeAssistant, data: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Validate the user input allows us to connect."""
+
+        # TODO: Some fancy validation of the selectors?
+        return {"title": data["name"]}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the initial step."""
+        errors = None
+        if user_input is not None:
+            errors = {}
+            try:
+                info = await self.validate_input(self._handler.hass, user_input)
+            except Exception:
+                # TODO: Error handling
+                errors["base"] = "blabla"
+            else:
+                return self._handler.async_create_entry(
+                    title=info["title"], data=user_input
```

Comment:
Not sure if the settings should be stored in `config_entry.data` or `config_entry.options`?

Reply:
Should it be `options` or else the options flow won't work?

Reply:
Shouldn't all data from the flow be stored as options ?